### PR TITLE
Feature/check why sending scan failed #14

### PR DESCRIPTION
**Pull request by slaible**

### CODE REVIEW

**Comment by slaible** [comment]
Signed-off-by: Stefan Laible [stefan.laible@de.bosch.com](mailto:stefan.laible@de.bosch.com)

File: bosch_locator_bridge/include/bosch_locator_bridge/locator_bridge_node.hpp [hunk 1]
```diff
@@ -65,6 +65,9 @@ class LocatorBridgeNode : public rclcpp::Node
   bool check_module_versions(
     const std::unordered_map<std::string, std::pair<int32_t,
     int32_t>> & module_versions);
+  bool get_config_entry(const std::string & name, std::string & value) const;
```
Comment (kfabian) [inline bosch_locator_bridge/include/bosch_locator_bridge/locator_bridge_node.hpp]: Might it make sense to have this templated to for completness with set_config_entry? There is a Poco::Dynamic::Var::convert method which could be used.

File: bosch_locator_bridge/include/bosch_locator_bridge/sending_interface.hpp [hunk 2]
```diff
@@ -35,10 +35,12 @@ class SendingInterface : public Poco::Runnable
   void run();
   virtual ~SendingInterface();
 
+  enum SendingStatus {SUCCESS, NO_CONNECTIONS, NOT_COMPLETED, RESET_EXCEPTION, IO_EXCEPTION};
```
Comment (kfabian) [inline bosch_locator_bridge/include/bosch_locator_bridge/sending_interface.hpp]: Using enum class over a plain enum is preferred.

File: bosch_locator_bridge/src/locator_bridge_node.cpp [hunk 3]
```diff
@@ -285,6 +285,36 @@ bool LocatorBridgeNode::check_module_versions(
...
+  } catch (const Poco::NotFoundException & error) {
+    RCLCPP_ERROR_STREAM(get_logger(), "Could not find config entry " << name << ".");
+    return false;
+  }
+
+  return true;
+}
+
+template<typename T>
+bool LocatorBridgeNode::set_config_entry(const std::string & name, const T & value) const
```
Comment (kfabian) [inline bosch_locator_bridge/src/locator_bridge_node.cpp]: A function template must be defined in a header file.

**Review by kfabian — CHANGES_REQUESTED** [review]

**Pull request merged**